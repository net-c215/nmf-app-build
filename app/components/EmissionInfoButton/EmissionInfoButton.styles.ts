import { StyleSheet } from "react-native";

import { Colors } from "../../style";

export default StyleSheet.create({
  infoIcon: {
    marginRight: 10,
    color: Colors.grey100,
  },
});
