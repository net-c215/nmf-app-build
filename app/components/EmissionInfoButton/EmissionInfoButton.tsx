import React from "react";
import { useSelector } from "react-redux";
import { pathOr } from "ramda";
import { Ionicons } from "@expo/vector-icons";
import { useNavigation, useRoute } from "@react-navigation/native";

import { emissions } from "../../ducks";
import { navigate } from "../../navigation";
import styles from "./EmissionInfoButton.styles";

const EmissionInfoButton = () => {
  const navigation = useNavigation();
  const navigator = navigate(navigation);

  const route = useRoute();
  const emissionId = pathOr("", ["params", "id"], route);

  const { emissionType } =
    useSelector((state) =>
      emissions.selectors.getEmissionById(state, emissionId)
    ) || {};

  return (
    <Ionicons
      name="md-information-circle"
      size={26}
      style={styles.infoIcon}
      onPress={() => navigator.openHtmlView({ emissionType: emissionType })}
    />
  );
};

export default EmissionInfoButton;
