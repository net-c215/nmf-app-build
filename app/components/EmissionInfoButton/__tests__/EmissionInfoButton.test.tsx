import React from "react";
import { create } from "react-test-renderer";

import EmissionInfoButton from "../";

it("renders correctly full width Primary Button with Primary font", () => {
  const tree = create(<EmissionInfoButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
