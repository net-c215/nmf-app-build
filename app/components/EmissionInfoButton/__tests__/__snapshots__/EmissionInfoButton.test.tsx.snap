// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly full width Primary Button with Primary font 1`] = `
<Ionicons
  name="md-information-circle"
  onPress={[Function]}
  size={26}
  style={
    Object {
      "color": "#111714",
      "marginRight": 10,
    }
  }
/>
`;
